src/backend/distributed/metadata/dependency.c

```diff
@@ -635,6 +635,10 @@ ApplyAddToDependencyList(ObjectAddressCollector *collector, Form_pg_depend pg_de
 	{
 		return;
 	}
+	else if (CitusExtensionObject(&address))
```

Comment:
When in the `apply` function it will still resolve any dependencies the Citus extension might have which is undesirable here.